From: Matt Benjamin <mbenjamin@redhat.com>
Date: Mon, 6 Feb 2017 11:40:51 -0500
Subject: rgw: complete backport of "bi list --max-entries=1" bugfix

Only one of 2 matching end-condition accounting changes was taken,
leading to radosgw-admin expecting an extra record.

Proximate cause of the backport issue was divergence of upstream
jewel from master in the second edited line--the change applied
cleanly but end result inconsistent.

Signed-off-by: Matt Benjamin <mbenjamin@redhat.com>
(cherry picked from commit 88bf9a047171e27988ac3879d2c925e84bfa994b)

Resolves: rhbz#1390716
---
 src/cls/rgw/cls_rgw.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cls/rgw/cls_rgw.cc b/src/cls/rgw/cls_rgw.cc
index 58247b5..67037a3 100644
--- a/src/cls/rgw/cls_rgw.cc
+++ b/src/cls/rgw/cls_rgw.cc
@@ -2517,7 +2517,7 @@ static int rgw_bi_list_op(cls_method_context_t hctx, bufferlist *in, bufferlist
 #define MAX_BI_LIST_ENTRIES 1000
   int32_t max = (op.max < MAX_BI_LIST_ENTRIES ? op.max : MAX_BI_LIST_ENTRIES) + 1; /* one extra entry for identifying truncation */
   string start_key = op.marker;
-  int ret = list_plain_entries(hctx, op.name, op.marker, max, &op_ret.entries) + 1; /* one extra entry for identifying truncation */
+  int ret = list_plain_entries(hctx, op.name, op.marker, max, &op_ret.entries);
   if (ret < 0) {
     CLS_LOG(0, "ERROR: %s(): list_plain_entries retured ret=%d", __func__, ret);
     return ret;
